src/compiler/compile/nodes/Element.ts

```diff
@@ -626,24 +615,11 @@ export default class Element extends Node {
 		}
 
 		if (this.name === 'label') {
-			const has_input_child = (children: INode[]) => {
-				if (children.some(child => (child instanceof Element && (a11y_labelable.has(child.name) || child.name === 'slot')))) {
-					return true;
-				}
-
-				for (const child of children) {
-					if (!('children' in child) || child.children.length === 0) {
-						continue;
-					}
-					if (has_input_child(child.children)) {
-						return true;
-					}
-				}
-
-				return false;
-			};
+			const rule_options = component.compile_options.a11y?.rules?.['label-has-associated-control'];
+			// magic number inspired from https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/src/rules/label-has-associated-control.js
+			const recursion_depth = Math.min(rule_options?.depth ?? 3, 25);
```

Comment:
[imo]

How about moving this line into `may_contain_input_child`?
Then we can prevent writing this line twice at this file and `src/compiler/compile/nodes/InlineComponent.ts`.